tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -274,4 +304,23 @@ static void setTrace(Trace trace) {
         MDC.put(Tracers.TRACE_ID_KEY, trace.getTraceId());
     }
 
+    protected static void checkForRemotingTracerCompatibility(Optional<OrderableSlsVersion> remotingTracerVer) {
+        if (remotingTracerVer.isPresent()
+                && VersionComparator.INSTANCE.compare(remotingTracerVer.get(), MIN_COMPAT_REMOTING_TRACER_VER) < 0) {
+            throw new IllegalStateException(String.format(
```

Comment:
can we use `com.palantir.safe-logging:preconditions` and `Checks.checkState` ?

Reply:
i think this is more readable. also `remotingTracerVer.get()` is used to construct the error message, it makes sense to only do that when needed.